src/components/Reports/HealthIndicatorReport/HealthIndicatorWidget/HealthIndicatorWidget.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { render, waitFor } from '@testing-library/react';
+import { GqlMockedProvider } from '__tests__/util/graphqlMocking';
+import { HealthIndicatorWidget } from './HealthIndicatorWidget';
+import { HealthIndicatorWidgetQuery } from './HealthIndicatorWidget.generated';
+
+const accountListId = 'account-list-1';
+const setShowHealthIndicator = jest.fn();
+const mutationSpy = jest.fn();
+
+const healthIndicatorScore = {
+  id: '1',
+  overallHi: 90,
+  ownershipHi: 80,
+  consistencyHi: 70,
+  successHi: 60,
+  depthHi: 50,
+};
+
+interface ComponentsProps {
+  healthIndicatorData?: HealthIndicatorWidgetQuery['healthIndicatorData'];
+  showHealthIndicator?: boolean;
+}
+const Components = ({
+  healthIndicatorData = [],
+  showHealthIndicator = true,
+}: ComponentsProps) => (
+  <GqlMockedProvider<{ HealthIndicatorWidget: HealthIndicatorWidgetQuery }>
+    mocks={{
+      HealthIndicatorWidget: {
+        healthIndicatorData,
+      },
+    }}
+    onCall={mutationSpy}
+  >
+    <HealthIndicatorWidget
+      showHealthIndicator={showHealthIndicator}
+      accountListId={accountListId}
+      setShowHealthIndicator={setShowHealthIndicator}
+    />
+  </GqlMockedProvider>
+);
+
+describe('HealthIndicatorWidget', () => {
+  it('renders nothing when there is no data', async () => {
+    const { queryByText, container } = render(
+      <Components showHealthIndicator={false} />,
+    );
+
+    await waitFor(() => {
+      expect(mutationSpy).toHaveGraphqlOperation('HealthIndicatorWidget');
+    });
+
+    expect(setShowHealthIndicator).toHaveBeenCalledWith(false);
+    expect(container).toBeEmptyDOMElement();
```

Comment:
Nice! I recently needed to check whether the component returned null and didn't render anything. I didn't know about this, but it looks cleaner than what I did.

Reply:
Yeah, it was something new to me as well.